akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -61,7 +64,21 @@ private[akka] class DaemonMsgCreateSerializer(val system: ExtendedActorSystem) e
           .setClazz(props.clazz.getName)
           .setDeploy(deployProto(props.deploy))
         props.args map serialize foreach builder.addArgs
-        props.args map (a ⇒ if (a == null) "null" else a.getClass.getName) foreach builder.addClasses
+        props.args.map { a ⇒
+          val argClassName =
+            if (a == null) "null"
+            else {
+              val className = a.getClass.getName
+              if (scala212OrLater && a.getClass.isInstanceOf[Serializable] && className.contains("$Lambda$")) {
+                // The serialization of the parameters is based on passing class name instead of
```

Comment:
Ok, I think I understand: so here we prepare for the deserialization site to simply treat this as a Serializable, which will use java serialization, which will use the generated lambda deserialization methods that java generated for it...?